Library/Homebrew/cmd/reinstall.rb

```diff
@@ -2,6 +2,14 @@
 
 module Homebrew
   def reinstall
+    if !MacOS.can_build?
```

Comment:
Yeah, that's a good idea. I'll create `check_build_flags` in `FormulaInstaller`.